anndata/dev/multi_files/_anndataset.py

```diff
@@ -0,0 +1,670 @@
+                _X = _X.astype(self._dtypes["X"], copy=False)
+
+        self._X = _X
+
+        return _X
+
+    @property
+    def X(self):
+        _X = self._gather_X()
+
+        return self._convert_X(_X) if self._convert_X is not None else _X
+
+    @property
+    def layers(self):
+        self._lazy_init_attr("layers", set_vidx=True)
+        return self._layers_view
+
+    @property
+    def obsm(self):
+        self._lazy_init_attr("obsm")
+        return self._obsm_view
+
+    @property
+    def obs(self):
+        self._lazy_init_attr("obs")
+        return self._obs_view
+
+    @property
+    def obs_names(self):
+        return self.reference.obs_names[self.oidx]
+
+    @property
+    def var_names(self):
+        return self.reference.var_names[self.vidx]
+
+    @property
+    def shape(self):
+        return len(self.obs_names), len(self.var_names)
+
+    @property
+    def convert(self):
+        return self._convert
+
+    @convert.setter
+    def convert(self, value):
+        self._convert = value
+        self._convert_X = _select_convert("X", self._convert)
+        for attr in ATTRS:
+            setattr(self, f"_{attr}_view", None)
+
+    def __len__(self):
+        return len(self.obs_names)
+
+    def __getitem__(self, index: Index):
+        oidx, vidx = _normalize_indices(index, self.obs_names, self.var_names)
+        resolved_idx = self._resolve_idx(oidx, vidx)
+
+        return AnnDataSetView(self.reference, resolved_idx)
+
+    @property
+    def has_backed(self):
+        for i, adata in enumerate(self.adatas):
+            if adata.isbacked and self.adatas_oidx[i] is not None:
+                return True
+        return False
+
+    def __repr__(self):
+        n_obs, n_vars = self.shape
+        descr = f"AnnDataSetView object with n_obs × n_vars = {n_obs} × {n_vars}"
+        all_attrs_keys = self._view_attrs_keys.copy()
+        for attr in self._attrs:
+            all_attrs_keys[attr] = list(getattr(self.reference, attr).keys())
+        for attr, keys in all_attrs_keys.items():
+            if len(keys) > 0:
+                descr += f"\n    {attr}: {str(keys)[1:-1]}"
+        return descr
+
+    def to_adata(self, ignore_X=False, ignore_layers=False):
+        if ignore_layers or self.layers is None:
+            layers = None
+        else:
+            layers = self.layers.to_dict(use_convert=False)
+        obsm = None if self.obsm is None else self.obsm.to_dict(use_convert=False)
+        obs = (
+            None
+            if self.obs is None
+            else pd.DataFrame(self.obs.to_dict(use_convert=False))
+        )
+
+        if ignore_X:
+            X = None
+            shape = self.shape
+        else:
+            X = self._gather_X()
+            shape = None
+
+        adata = AnnData(X, obs=obs, obsm=obsm, layers=layers, shape=shape)
+        adata.obs_names = self.obs_names
+        adata.var_names = self.var_names
+        return adata
+
+
+class AnnDataSet(_ConcatViewMixin):
```

Comment:
Can we get a full spec of the docstring that clarifies all subtleties of the params?

Reply:
Most importantly, describe that `AnnDataSet` does not give access to most fields of `AnnData`, but `AnnDataSetView` does.